src/pretalx/mail/models.py

```diff
@@ -322,6 +323,11 @@ def send(self, requestor=None, orga: bool = True):
         )
 
         self.sent = now()
+        queuedmail_has_been_sent.send(
```

Comment:
Let's call the signal after the object has been saved, I'd say. Otherwise, this can easily lead to inconsistencies, and the signal name implies that the action has been completed.

Reply:
I agree, i was just confused by `self.save()` only getting called if `self.pk` evaluates to true. That's why this is here. I will move it to after the `if` clause.

Reply:
The idea is basically to allow the use of `QueuedMail` objects as interfaces for email sending (generating pretty HTML, event settings etc) but without adding those emails to the organiser-accessible log. This is e.g. used when sending emails to users with draft proposals, who should **not** be visible to organisers, ever!

Actually, due to this, we may want to only trigger the signal in the `if self.pk` block after all.

Reply:
Reviewing pretalx code, this hidden sending is used

- to send organisers informations ("News from your content system")
- to send out access codes
- when one speaker invites other speakers to their proposal (also nothing organisers ought to see!)
- when sending a notification to the organiser address that there has been a new proposal


So I'd say the signal can just go in the `pk=True` branch of the code, where it won't accidentally read hidden information. If a plugin really wants to see **all** outgoing emails, the easiest way would be to implement its own email backend (but then it would have to handle these distinctions by itself, especially the personal messages from speakers to other people, and the draft proposals, where pretalx makes guarantees of not leaking the identities and contents of non-submitted proposals).